SMLHelper/Assets/CustomFishPrefab.cs

```diff
@@ -0,0 +1,132 @@
+            rb.angularDrag = 1f;
+
+            WorldForces forces = mainObj.GetOrAddComponent<WorldForces>();
+            forces.useRigidbody = rb;
+            forces.aboveWaterDrag = 0f;
+            forces.aboveWaterGravity = 9.81f;
+            forces.handleDrag = true;
+            forces.handleGravity = true;
+            forces.underwaterDrag = 1f;
+            forces.underwaterGravity = 0;
+            forces.waterDepth = Ocean.main.GetOceanLevel();
+            forces.enabled = false;
+            forces.enabled = true;
+
+            mainObj.GetOrAddComponent<EntityTag>().slotType = EntitySlot.Type.Creature;
+            mainObj.GetOrAddComponent<PrefabIdentifier>().ClassId = ClassID;
+            mainObj.GetOrAddComponent<TechTag>().type = TechType;
+
+            mainObj.GetOrAddComponent<SkyApplier>().renderers = renderers;
+            mainObj.GetOrAddComponent<LargeWorldEntity>().cellLevel = LargeWorldEntity.CellLevel.Near;
+            mainObj.GetOrAddComponent<LiveMixin>().health = 10f;
+
+            Creature creature = mainObj.GetOrAddComponent<Creature>();
+            creature.initialCuriosity = AnimationCurve.Linear(0f, 0.5f, 1f, 0.5f);
+            creature.initialFriendliness = AnimationCurve.Linear(0f, 0.5f, 1f, 0.5f);
+            creature.initialHunger = AnimationCurve.Linear(0f, 0.5f, 1f, 0.5f);
+            SwimBehaviour behaviour = null;
+            if (isWaterCreature)
+            {
+                behaviour = mainObj.GetOrAddComponent<SwimBehaviour>();
+                SwimRandom swim = mainObj.GetOrAddComponent<SwimRandom>();
+                swim.swimVelocity = swimSpeed;
+                swim.swimRadius = swimRadius;
+                swim.swimInterval = 1f;
+            }
+            else
+            {
+                behaviour = mainObj.GetOrAddComponent<WalkBehaviour>();
+                WalkOnGround walk = mainObj.GetOrAddComponent<WalkOnGround>();
+                OnSurfaceMovement move = mainObj.GetOrAddComponent<OnSurfaceMovement>();
+                move.onSurfaceTracker = mainObj.GetOrAddComponent<OnSurfaceTracker>();
+            }
+            Locomotion loco = mainObj.GetOrAddComponent<Locomotion>();
+            loco.useRigidbody = rb;
+            mainObj.GetOrAddComponent<EcoTarget>().type = EcoTargetType.Peeper;
+            mainObj.GetOrAddComponent<CreatureUtils>();
+            mainObj.GetOrAddComponent<VFXSchoolFishRepulsor>();
+            SplineFollowing spline = mainObj.GetOrAddComponent<SplineFollowing>();
+            spline.locomotion = loco;
+            spline.levelOfDetail = mainObj.GetOrAddComponent<BehaviourLOD>();
+            spline.GoTo(mainObj.transform.position + mainObj.transform.forward, mainObj.transform.forward, 5f);
+            behaviour.splineFollowing = spline;
+
+            if (pickupable)
+            {
+                Console.WriteLine("[FishFramework] Adding pickupable component");
+                mainObj.GetOrAddComponent<Pickupable>();
```

Comment:
If the fish can be picked up, then it can be turned into an `InventoryItem`.  
That means that this entity will need a TechType and an icon for use in containers.

Take a look at the [Spawnable](https://github.com/SMLHelper/SMLHelper/blob/2.x/SMLHelper/Assets/Spawnable.cs) class that inherits from `ModPrefab` and see if there is anything there that can help with ensuring that a custom fish that is picked up will work as intended.

If this is somehow already working, disregard this comment.

Reply:
The player can pick up the fish just fine, but I have yet to integrate icons

Reply:
It can be left to the modder to register a sprite for their new fish `TechType`